src/test/java/com/google/cloud/tools/maven/util/SingleYamlDeployTestHelper.java

```diff
@@ -0,0 +1,97 @@
+package com.google.cloud.tools.maven.util;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.cloud.tools.appengine.api.deploy.AppEngineDeployment;
+import com.google.cloud.tools.appengine.api.deploy.AppEngineFlexibleStaging;
+import com.google.cloud.tools.appengine.api.deploy.AppEngineStandardStaging;
+import com.google.cloud.tools.maven.AbstractSingleYamlDeployMojo;
+import com.google.cloud.tools.maven.CloudSdkAppEngineFactory;
+import com.google.common.base.Charsets;
+import com.google.common.io.Files;
+
+import org.junit.rules.ExternalResource;
+import org.junit.rules.TemporaryFolder;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.io.File;
+import java.io.IOException;
+
+public class SingleYamlDeployTestHelper<M extends AbstractSingleYamlDeployMojo>
```

Comment:
External Resource thing is cool, would it make sense to have this as separate flex/standard fixtures?, it does seem like most of the heavy lifting is in the branches in the before and after?